test-violations.py

```diff
@@ -0,0 +1,62 @@
+from attrs import define
+from attrs import field
+
+
+@attr.s(auto_attribs=True)
+class ExampleClass(object):
+    # ruleid:attr-mutable-initializer
+    empty_dict = {}
+    # ruleid:attr-mutable-initializer
+    empty_list = []
+    # ruleid:attr-mutable-initializer
+    somedict = dict()
+    # ruleid:attr-mutable-initializer
+    somelist = list()
+    # ruleid:attr-mutable-initializer
+    someset = set()
+    # ok:attr-mutable-initializer
+    ex_good1 = attr.ib(factory=dict)
+    # ok:attr-mutable-initializer
+    ex_good2 = attr.ib(factory=list)
+    # ok:attr-mutable-initializer
+    ex_good3 = attr.ib(factory=set)
+    # ruleid:attr-mutable-initializer
+    myset = {1, 2, 3}
+
+    def foo(self):
+        # ok:attr-mutable-initializer
+        x = {}
+
+    def bar(self) -> int:
+        # ok:attr-mutable-initializer
+        thisset = set()
+
+
+@define
+class ExampleClass(object):
+    # ruleid:attr-mutable-initializer
+    empty_dict = {}
+    # ruleid:attr-mutable-initializer
+    empty_list = []
```

Comment:
:exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium
 <!-- 49d6e81a-f6c0-4894-ae75-1425ca2d7e47 -->